In [20]:
# Required libraries
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns
import ast
import warnings
warnings.filterwarnings('ignore')

In [5]:
chipo = pd.read_csv("D:\Desktop\Chipotle\chipotle.csv", sep="\t")

In [ ]:
# Checking first 10 row of our data
chipo.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

## What is the number of observations in the dataset?

In [ ]:
# Checking the rows in our data 
chipo.shape[0] 

4622

In [8]:
# Checking the columns in our data 
chipo.shape[1]

5

In [11]:
# Printing the columns name 
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [14]:
chipo.isnull().sum()

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

### Observation 

We don't have null values whatsoever but there is some values missing in "choice_description"
Which cant be treated as Null values they are just customer prefrences to not add any choice 
over there food

## Which was the most-ordered item?


In [ ]:
# Checking the most orderd dish in our menu 
chipo['item_name'].value_counts().head(5) 

Chicken Bowl           726
Chicken Burrito        553
Chips and Guacamole    479
Steak Burrito          368
Canned Soft Drink      301
Name: item_name, dtype: int64

### Observation 

1) Above we get top 5 most-ordered item

2) The most-ordered item in our data is "Chicken Bowl"

3) For the most oredered item chicken bowl oredered about 726 units


#### ~



### What was the most ordered item in the choice_description column?


In [ ]:
# choice_description is unstructered
chipo['choice_description'].values[5]

'[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]'

In [ ]:
# Before moving further we have to fix the col using ast
# Below code ensures we skip missing values and make literal or list of list or just list

Choice_des = []

for i in chipo['choice_description']:
    if pd.notna(i): 
        try:
            Choice_des.append(ast.literal_eval(i))

        except (ValueError,SyntaxError):
            Choice_des.append([i])
    else:
        Choice_des.append('No Choice')

chipo['AddOn_choice']=Choice_des
chipo.head()
# print(Choice_des[:10])

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  \
0                                                NaN     $2.39    
1                                       [Clementine]     $3.39    
2                                            [Apple]     $3.39    
3                                                NaN     $2.39    
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98    

                                        AddOn_choice  
0                                          No Choice  
1                                     [[Clementine]]  
2                                          [[Apple]]  
3                                          No Choice  
4  [[Tomatillo-Red Chili Salsa (Hot), [Black Bean...

In [ ]:
# We can drop choice_description 
chipo = chipo.drop(columns=['choice_description'])
chipo.head(3)

In [ ]:
# WE can now use explode method to make list in individual row and can check 

most_common_choice = (
    chipo.explode('AddOn_choice')['AddOn_choice'].value_counts().head(5)
)

print(most_common_choice)

No Choice                                                                 1246
[Diet Coke]                                                                134
[Coke]                                                                     123
[Sprite]                                                                    77
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Lettuce]]      42
Name: AddOn_choice, dtype: int64


## Observation

1) Most of the customer go without any addons 
2) But we can say Diet Coke is the most orederd item in addons
3) Top 3 are coldrinks diet coke, Coke, Sprite

# ~



## How many items were orderd in total?

In [ ]:
# Lets check detailed items and counts 
chipo.groupby('item_name').sum()

order_id  quantity
item_name                                                
6 Pack Soft Drink                         52322        55
Barbacoa Bowl                             53972        66
Barbacoa Burrito                          74718        91
Barbacoa Crispy Tacos                      5613        12
Barbacoa Salad Bowl                        9708        10
Barbacoa Soft Tacos                       18725        25
Bottled Water                            175944       211
Bowl                                        472         4
Burrito                                    1550         6
Canned Soda                               76396       126
Canned Soft Drink                        304753       351
Carnitas Bowl                             62742        71
Carnitas Burrito                          51652        60
Carnitas Crispy Tacos                      5868         8
Carnitas Salad                             1500         1
Carnitas Salad Bowl                        6693         6
Carnitas Soft Tacos                       37673        40
Chicken Bowl                             713926       761
Chicken Burrito                          497303       591
Chicken Crispy Tacos                      47382        50
Chicken Salad                             10465         9
Chicken Salad Bowl                       117104       123
Chicken Soft Tacos                        98395       120
Chips                                    208004       230
Chips and Fresh Tomato Salsa             100419       130
Chips and Guacamole                      449959       506
Chips and Mild Fresh Tomato Salsa           279         1
Chips and Roasted Chili Corn Salsa        14999        23
Chips and Roasted Chili-Corn Salsa        17462        18
Chips and Tomatillo Green Chili Salsa     41170        45
Chips and Tomatillo Red Chili Salsa       45556        50
Chips and Tomatillo-Green Chili Salsa     21826        33
Chips and Tomatillo-Red Chili Salsa       18344        25
Crispy Tacos                                434         2
Izze                                      10587        20
Nantucket Nectar                          21814        29
Salad                                      1150         2
Side of Chips                             84769       110
Steak Bowl                               193752       221
Steak Burrito                            328437       386
Steak Crispy Tacos                        35818        36
Steak Salad                                2362         4
Steak Salad Bowl                          29590        31
Steak Soft Tacos                          50177        56
Veggie Bowl                               72270        87
Veggie Burrito                            80962        97
Veggie Crispy Tacos                         668         1
Veggie Salad                               7265         6
Veggie Salad Bowl                         17303        18
Veggie Soft Tacos                          5520         8

In [37]:
# Calculating total item solds as per our unit 

chipo['quantity'].sum()

4972

## Observation

Totaly we sold 4972 items including cold drinks , dish etc. But Chicken bowl is the star dish

# ~
#### Turn the item price into a float

In [39]:
# Observation:- We cant directly convert price into float because it has ($) sign 
# We have to get rid of that $ sign then we can convert it into a float
# Below code will match pattern and rid of $ sign and convert it into float
chipo['item_price']= chipo['item_price'].str.replace('$',"",regex=True).astype(float)

print(chipo['item_price'].head(3))
print(chipo['item_price'].dtype)

0    2.39
1    3.39
2    3.39
Name: item_price, dtype: float64
float64


### How much was the revenue for the period in the dataset?

In [ ]:
# Lets calculate the revenue we made
chipo['item_price'].sum()
# We have generated revenue around $34500.16 

34500.16

### How many orders were made in the period?

In [ ]:
# Lets count how many oreder we completed 
chipo['item_name'].count()
# We have made around 4622 orders

4622

### What is the average revenue amount per order?

In [ ]:
# Lets check how much customer are paying averagly 
# Before that lets check if there are any outlier in price column
# According to that we can decide which mehtod we will go for (mean,median)
chipo['item_price'].describe()

count    4622.000000
mean        7.464336
std         4.245557
min         1.090000
25%         3.390000
50%         8.750000
75%         9.250000
max        44.250000
Name: item_price, dtype: float64

In [ ]:
# looks like some amount are extreme in price col let's use median 
chipo['item_price'].median()
# So averagely our customer are paying 8.75 $ 

8.75

### How many different items are sold?

In [ ]:
# Lets find out how many diffrent items are sold
unique_items = chipo['item_name'].unique()
print(len(unique_items))
# There are 50 diffrent items which are sold


50


# Data Visualisation

### Create a bar chart of top 10 most-ordered items.


In [48]:
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'item_price', 'AddOn_choice'], dtype='object')

In [ ]:
top_10=(
    chipo.groupby('item_name')['quantity']
    .sum()
    .sort_values(ascending=False)
    .head(10)
)

sns.barplot(data=top_10, x='quantity', y='item_name', palette="viridis")
plt.title("Top 10 Most-Ordered Items")
plt.xlabel("Total Quantity Ordered", fontsize=12)
plt.ylabel("Item Name", fontsize=12)
plt.tight_layout()
plt.show()

ValueError: Could not interpret input 'item_name'